plasmapy/diagnostics/thomson.py

```diff
@@ -111,12 +129,39 @@ def spectral_density(
 
     electron_vel : `~astropy.units.Quantity`, shape (Ne, 3), optional
         Velocity of each electron component in the rest frame. (convertible to m/s)
+        If set, overrides electron_vdir and electron_speed.
         Defaults to a stationary plasma [0, 0, 0] m/s.
 
+    electron_vdir : np.ndarray, shape (Ne,3), optional
+        Unit vectors describing the velocity of each electron population.
+        Unit vectors will be normalized, but must not contain all zero elements.
+        Setting electron_vel overrides this keyword.
+
+    electron_speed : `~astropy.units.Quantity`, shape (Ne), optional
+        A scalar speed for each electron population. Must be used along with
```

Comment:
I see you switch back and forth between using "component" and "population" in the docstrings.  On an initial read, I think users might interpret these as two different things, so, to reduce confusion, I would stick with just one.  I'll leave the final decision up to you, but I think "population" would be the better route since the input arguments are properties associated with the different electron/ion populations of the distribution.

Reply:
Good point - done.